src/paymentdisclosure.h

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef ZCASH_PAYMENTDISCLOSURE_H
+#define ZCASH_PAYMENTDISCLOSURE_H
+
+#include "uint256.h"
+#include "clientversion.h"
+#include "serialize.h"
+#include "streams.h"
+#include "version.h"
+
+// For JSOutPoint
+#include "wallet/wallet.h"
+
+#include <cstdint>
+#include <string>
+
+
+// Ensure that the two different protocol messages, payment disclosure blobs and transactions,
+// which are signed with the same key, joinSplitPrivKey, have disjoint encodings such that an
+// encoding from one context will be rejected in the other.  We know that the set of valid
+// transaction versions ({1..INT32_MAX}) so we will use a negative value for payment disclosure.
+// The negative value -9411486 in hex is 0xFF706462 and in ascii ÿpdb.
```

Comment:
That's not strictly speaking ASCII, it's ISO-8859-1.
Also the {1..INT32_MAX} is only the set of *currently* valid transaction versions; it will likely change in NU0, but the spec will reserve -2<sup>24</sup>..-1 for magic values.

Reply:
Updating comment.

Reply:
Wait, -9411486 is hex FF 70 64 62 in *big* endian (in little-endian it would be "bdpÿ" as ISO-8859-1).

Reply:
This comment has now been addressed.